docs/COMPATIBILITY.md

```diff
@@ -0,0 +1,17 @@
+# Compatibility
+
+Our goal as Statesman maintainers is for the library to be compatible with all supported versions of Ruby and Rails.
+
+Specifically, any CRuby/MRI version that has not received an End of Life notice ([e.g. this notice for Ruby 2.1](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/)) is supported. Similarly, any version of Rails listed as currently supported on [this page](http://guides.rubyonrails.org/maintenance_policy.html) is one we aim to support in Statesman.
+
+To that end, [our build matrix](.travis.yml) includes all these versions.
+
+Any time Statesman doesn't work on a supported combination of Ruby and Rails, it's a bug, and can be reported [here](https://github.com/gocardless/statesman/issues).
+
+# Deprecation
+
+Whenever a version of Ruby or Rails falls out of support, we will mirror that change in Statesman by updating the build matrix and releasing a new major version.
+
+At that point, we will close any issues that only affect the unsupported version, and may choose to remove any workarounds from the code that are only necessary for the unsupported version.
+
+We will then bump the major version of Statesman, to indicate the break in compatibility. Even if the new version of Statesman happens to work on the unsupported version of Ruby or Rails, we consider compatibility to be broken at this point.
```

Comment:
> You can imagine a situation in which there are no actual changes to functionality or compatibility, but we just release an "empty" version.

So this is exactly what we're thinking of doing.

Our reasoning is that once you take those versions out of CI, you have no idea if you work with them any more. We think that a bump of the major version is the indicator most likely to get users to take stock and check the changelog.

The other options we discussed - and they're still on the table - were:

1) Publish the policy, roll Ruby/Rails out of support without releasing any kind of new version
2) Publish the policy, allow breaking compatibility in minor releases (not patch)

The reason we dismissed those was that we figured people don't go and read the compatibility policies of every library they use, and were unlikely to realise that we'd dropped support. The one thing they'd be most likely to see is the bump in the major version, which would prompt them to go investigate why it was bumped.

Honestly we're a little torn, but this seemed like the option with the best outcome for everyone. It definitely feels weird to rattle through major releases like this though, but I can't back up that feeling with an actual reason for it being bad.

I'm going to see if some friends with experience maintaining open source projects have any thoughts. Don't be surprised if some new people show up in this thread!